compiler/src/dmd/safe.d

```diff
@@ -373,6 +374,22 @@ extern (D) bool setUnsafe(
     return false;
 }
 
+extern (D) bool setFunctionToUnsafe(FuncDeclaration fd)
+{
+    if (fd.safetyInprocess)
+    {
+        fd.safetyInprocess = false;
+        fd.type.toTypeFunction().trust = TRUST.system;
+
+        if (fd.fes)
+            setFunctionToUnsafe(fd.fes.func);
```

Comment:
I understand the renaming, but I don't think it's a good idea to duplicate this function with partially evaluated arguments: it only adds new confusing overloading, and every call to this function is basically a diagnostic bug and should be replaced with a call to the first overload of `setFunctionToUnsafe`, which explains to the user why a function is being marked unsafe.

Reply:
The problems are:

1. the multiple different overloadings of setUnsafe
2. the tangle of different behaviors taken based on whether fmt is null or arg is null or both are null
3. fhe absurdly complicated forwarding of the optional arguments

I've spent far too much time trying to trace through the logic of this function and its partner `errorSupplementalInferredAttr()`

My plan is to replace this rube goldberg machine with proper use of `va_list`. Problem (2) should split the function into 3 distinct functions. This particular PR factors out the null null case.

It's my fault that va_list wasn't used in the original incarnation of this, and I aim to fix it. The first two steps are done - this one, and improving ErrorSink to support va_list.

Reply:
The solution to the `arg ? toChars(arg) : null` inefficiency is to split the function into two:

1. determine if there is an error
2. if so, then produce the error message

AttributeViolation can use OutBuffer.printf to produce a string, as you are correct in that the `va_list` cannot be saved.